.github/workflows/publish.yml

```diff
@@ -1,22 +1,35 @@
 name: publish
 
-on: [push]
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - 'main'
+    tags:
+      - 'v*'
+  pull_request:
```

Comment:
Is this `pull_request` key necessary - shouldn't it be canceled out by `push: ${{ github.event_name != 'pull_request' }}` on line 32?